### PR TITLE
Feature: Implement iCal integration for calendar/agenda, improve calendar/agenda styling

### PR DESCRIPTION
**Pull request by denispapec**
## Proposed change

This PR implements iCal integration for calendar/agenda. 
Having iCal events in calendar/agenda, new configuration setting was added to show time in a day for events happening today.
During the implementation some styling improvements were made to bring styling consistency in line with other widgets, and event line was simplified and is now shared between calendar and agenda, having the same functionality.

Closes #1294 
Closes #2263 

## Type of change

- [ ] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [x] Other (please explain): New calendar/agenda integration

## Checklist:

- [ ] If adding a service widget or a change that requires it, I have added corresponding documentation changes.
- [ ] If adding a new widget I have reviewed the [guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [x] I have checked that all code style checks pass using pre-commit hooks and linting checks with `pnpm lint` (see development guidelines).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

**Review by shamoon — COMMENTED** [review]
Nice work, in general this works well. To be honest, Im a little worried about the potential for numerous little FRs given how particular calendar stuff can be, but lets hope not and we'll aim to keep this dead-simple.

I like the refactoring and getting rid of the context provider (I removed the actual file)

My only significant-ish issue was with the changes to the generic proxy handler, that felt kind of 'shoehorned'. I also didnt like that there was a calendar 'widget' and ical 'widget that seemed a bit redundant. Instead I created a generic proxy for the calendar component and linked ical to that. It keeps the code tidy and the integration stuff out of generic proxy handler, which may also prove helpful going forward.

Other stuff was minor, such as adding a little error if no events are found (presumably an error, like if you use google.com as your ical URL).

As always, let me know your thoughts.

Screenshot 2023-11-24 at 11 32 26 PM

File: src/widgets/calendar/integrations/ical.jsx [hunk 1]
```diff
@@ -10,13 +10,21 @@ export default function Integration({ config, params, setEvents, hideErrors }) {
...
   });
 
+  let parsedIcal;
+
+  if (!icalError && icalData && !icalData.error) {
+    parsedIcal = parseString(icalData.data);
+    if (parsedIcal.events.length === 0) {
+      icalData.error = { message: "No events found" };
+    }
+  }
```
Comment (denispapec) [inline src/widgets/calendar/integrations/ical.jsx:20]: Seems like this change causes continuous re-rendering. Reverting this change back solves this issue. Is there maybe a different way of sorting this one?

**Comment by denispapec** [comment]
In the beginning I did create new proxy handler, but seeing how minimal changes are needed for generic handler, I've decided to extend it,  haha :sweat_smile: . It does make more sense having a separate handler :+1: .

I can see few more simple changes that can be added: 
- translation for `No events found`
- add ical name (`config.name`) into the error message, so in the error response it can say: `API error: ical (google): No events found.` 

Let me know if you want me to make changes for these two suggestions.

**Comment by shamoon** [comment]
Thanks, good catch, didnt meant to put that outside the useEffect hook =)

Good suggestions too, added those. Let me know if you have any other concerns / thoughts.

**Comment by denispapec** [comment]
Nice, thanks.
LGTM

**Review by shamoon — APPROVED** [review]
Thanks again

**Pull request merged**